packages/rocketchat-livechat/.app/client/lib/chatMessages.js

```diff
@@ -142,7 +142,7 @@ this.ChatMessages = class ChatMessages {
 			});
 		};
 
-		if (!visitor.getId()) {
+		if ((!visitor.getId()) || (visitor.getDepartment() !== Livechat.department)) {
```

Comment:
please avoid unnecessary parenthesis
```suggestion
		if (!visitor.getId() || visitor.getDepartment() !== Livechat.department) {
```